packages/legend-graph/src/graphManager/protocol/pure/v1/transformation/pureGraph/to/helpers/V1_DomainBuilderHelper.ts

```diff
@@ -262,6 +292,11 @@ export const V1_buildAssociationProperty = (
     associatedProperty.type,
     `Association associated property 'type' field is missing`,
   );
+  validateAssociationProperty(
+    pureAssociation,
+    associatedPropertyClassType,
+    context,
+  );
   const associatedClass = context.resolveClass(associatedPropertyClassType);
   const property = V1_buildProperty(currentProperty, context, pureAssociation);
   associatedClass.value.propertiesFromAssociations.push(property);
```

Comment:
Would it better to validate once the property has been built.
If the property type is in system then its invalid ?

Reply:
I think we should do it before building the graph Because these validations are being made while we build the graph so it's better to check if it is not present system first and build the property